source/draw/gpu/opengl/backend/gles3/Gles3ShaderProgram.ooc

```diff
@@ -86,6 +87,31 @@ Gles3ShaderProgram: class extends GLShaderProgram {
 		glUniform4f(glGetUniformLocation(this _backend, name), value x, value y, value z, value w)
 		version(debugGL) { validateEnd("ShaderProgram setUniform~FloatPoint4D") }
 	}
+	setUniform: override func ~ColorBgr (name: String, value: ColorBgr) {
+		version(debugGL) { validateStart("ShaderProgram setUniform~ColorBgr") }
+		glUniform3f(glGetUniformLocation(this _backend, name), (value blue as Float) / 255.0f, (value green as Float) / 255.0f, (value red as Float) / 255.0f)
```

Comment:
Btw, `FloatPoint4D` apparently has an invalid default constructor: `	init: func@ ~default { this init(0.0f, 0.0f, 0.0f, 1.0f) }`

Apparently you shouldn't assign anything to anything but 0 in default constructors for covers, because if you create an array of these things, you should be able to assume that they have all been "constructed", but they will be in an invalid state (alpha will be 0) if you do create an array and the default constructor looks like this. Something like that.